File: RennDscx.m
%
% Wrapper function for program dscx
%
% IN    pthImg  image file path
%       pthOut  output file path
%       Admin   administration, u_CmndAdmin.m
%
% OUT   Out     standard output
% 
function [Out] = RennDscx( pthImg, pthOut, Admin )

if nargin==2, 
    Admin.pthProg = ''; 
    Admin.optS    = '';
end

cmnd  	= [Admin.pthProg 'dscx ' pthImg ' ' pthOut ' ' Admin.optS ];

[status Out] = dos(cmnd);                   % excecute program

%% ------  Status  ------
if status>0
    Out
    warning('something went wrong');
end

%% ------  Verify Proper Termination  -----
ixEOP = strfind( Out, 'EndOfProgram');

if isempty(ixEOP)
    warning('Command %s not executed.', cmnd);
    Out
    fprintf('Paused');
    pause();
end

File: RennDscx.py
#
# Wrapper function for program dscx
#
# IN    pthImg  image file path
#       pthOut  output file path
#       Admin   administration, u_ProgExecAdmin.m
#
# OUT   Out     standard output
#
import subprocess

def RennDscx( pthImg, pthOut, Admin ):

    # https://docs.python.org/3/library/subprocess.html
    # python 3.6
    cmd  = 'dscx ' + pthImg + ' ' + pthOut + ' ' + Admin.optS
    #Args = subprocess.run( [ "dscx", pthImg, pthOut, Admin.optS ],
    Args = subprocess.run( cmd,
			   stdout = subprocess.PIPE,
			   stderr = subprocess.PIPE,
			   universal_newlines = True )
    
    print( Args.stdout )
    print( Args.stderr )

    # ------  Verify Proper Termination  -----
    ixEOP = Args.stdout.find('EndOfProgram')
    if ixEOP<0:
        print('Program dscx was not executed. Output below:' )
        print( Admin.optS )
        print( Args.stdout )
        quit()

    return Args.stdout


